flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -1389,6 +1388,149 @@ private void copyStateDataHandleData(
 		return Tuple2.of(stateInfo.f0, newMetaInfo);
 	}
 
+	private <N, S extends State, SV> RegisteredKeyValueStateBackendMetaInfo<N, SV> migrateStateIfNecessary(
```

Comment:
I have an ongoing effort to shrink the size of the `RocksDBKeyedStateBackend` class. I think it is a good idea to move the migration concern in a separate class file if possible to keep different concerns more separated whereever possible.

Reply:
Thoughs about this point?